code/modules/projectiles/gun_attachables.dm

```diff
@@ -1322,14 +1323,16 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	if(get_dist(user,target) > max_range)
 		to_chat(user, span_warning("Too far to fire the attachment!"))
 		return
-	if(current_rounds > 0)
+	if(current_rounds > 0 && COOLDOWN_CHECK(src, last_fired))
 		prime_grenade(target,gun,user)
 
 
 /obj/item/attachable/attached_gun/grenade/proc/prime_grenade(atom/target, obj/item/weapon/gun/gun, mob/living/user)
 	set waitfor = FALSE
 	var/nade_type = loaded_grenades[1]
 	var/obj/item/explosive/grenade/frag/G = new nade_type (get_turf(gun))
+	var/fire_delay = attachment_firing_delay
+	COOLDOWN_START(src, last_fired, fire_delay)
```

Comment:
Why is this a var

Reply:
This is an exact copy + paste of underbarrel flamer code. If you want me to change this then I'll change under barrel flamer code as well.

Reply:
Should be fixed now.